### PR TITLE
Build Bookmark and explore page

### PR DESCRIPTION
**Pull request by ishitakeshawani**
- An added feature of adding posts to bookmarks
- remove it from the bookmark
- Build the right section to home page
- Build explore page
- preview link: https://deploy-preview-4--todayspost.netlify.app/

### CODE REVIEW

File: src/components/leftsidebar/LeftSidebar.js [hunk 1]
```diff
@@ -34,16 +34,16 @@ export function LeftSidebar({
   ];
   return (
     <div className="left-section">
-      {sideBarItems.map(({ name, path, iconName }) => (
-        <NavLink
+      {sideBarItems.map(({ name, path, iconName },index) => (
+        <NavLink key={index}
```
Comment (surbhi84) [inline src/components/leftsidebar/LeftSidebar.js]: It is a better practice to add the key as some value that belongs to the array item as that won't change over time,while the index can change.
Good to know - 'Keys help React differentiate and distinguish elements from each other, increasing its performance when diffing between the virtual and real DOM.'
You can read more about why we use keys.

File: src/features/auth/authSlice.js [hunk 2]
```diff
@@ -43,6 +44,14 @@ export const login = createAsyncThunk(
 
 export const logout = createAction("auth/logout");
 
+export const getAllUsers = createAsyncThunk("auth/getAllUsers", async () => {
+  const {
+    data: { users },
+  } = await axios.get("/api/users");
+  console.log(users);
```
Comment (surbhi84) [inline src/features/auth/authSlice.js]: You can remove it if you don't need it anymore.

**Review by surbhi84 — APPROVED** [review]
- Liked that you used redux and thunk.
- The code looks clean, easily readable.
- Good to merge.

Do check the login test credentials once, didn't work for me.

File: src/components/leftsidebar/LeftSidebar.js (hunk 1 above)
Comment (hrishikeshmane) [reply src/components/leftsidebar/LeftSidebar.js]: You can use nanoid for keys- https://www.npmjs.com/package/nanoid

**Review by hrishikeshmane — APPROVED** [review]
Looks alright to me

File: src/components/leftsidebar/LeftSidebar.js (hunk 1 above)
Comment (ishitakeshawani) [reply src/components/leftsidebar/LeftSidebar.js]: Yes, I will read about it. and used nanoid here. Thank you for review @surbhi84 @hrishikeshmane

File: src/features/auth/authSlice.js (hunk 2 above)
Comment (ishitakeshawani) [reply src/features/auth/authSlice.js]: done

**Comment by ishitakeshawani** [comment]
> * Liked that you used redux and thunk.
> * The code looks clean, easily readable.
> * Good to merge.
> 
> Do check the login test credentials once, didn't work for me.

Yes, there are some changes to do in login. Completing it in the next PR.

**Pull request merged**